dex/src/main/scala/com/wavesplatform/dex/api/http/routes/v0/MarketRoute.scala

```diff
@@ -0,0 +1,392 @@
+import com.wavesplatform.dex.actors.address.AddressActor
+import com.wavesplatform.dex.api.http.directives.HttpKamonDirectives._
+import com.wavesplatform.dex.api.http.directives.ProtectDirective
+import com.wavesplatform.dex.api.http.entities._
+import com.wavesplatform.dex.api.http.headers.`X-User-Public-Key`
+import com.wavesplatform.dex.api.http.{HasStatusBarrier, OrderBookHttpInfo, _}
+import com.wavesplatform.dex.api.routes.PathMatchers.{AddressPM, AssetPairPM, OrderPM, PublicKeyPM}
+import com.wavesplatform.dex.api.routes.{ApiRoute, AuthRoute}
+import com.wavesplatform.dex.app.MatcherStatus
+import com.wavesplatform.dex.db.OrderDb
+import com.wavesplatform.dex.domain.account.{Address, PublicKey}
+import com.wavesplatform.dex.domain.asset.AssetPair
+import com.wavesplatform.dex.domain.order.Order
+import com.wavesplatform.dex.domain.utils.ScorexLogging
+import com.wavesplatform.dex.effect.FutureResult
+import com.wavesplatform.dex.model.{AssetPairBuilder, _}
+import com.wavesplatform.dex.queue.MatcherQueue.StoreValidatedCommand
+import com.wavesplatform.dex.queue.ValidatedCommand
+import com.wavesplatform.dex.settings.{OrderRestrictionsSettings}
+import io.swagger.annotations._
+
+import javax.ws.rs.Path
+import scala.concurrent.{ExecutionContext, Future}
+
+object MarketRoute {
+
+  case class Settings(
+    timeout: Timeout,
+    getActualTickSize: AssetPair => FutureResult[BigDecimal],
+    orderRestrictions: Map[AssetPair, OrderRestrictionsSettings]
+  )
+
+}
+
+@Path("/matcher")
+@Api()
+class MarketRoute(
+  settings: MarketRoute.Settings,
+  addressActor: ActorRef,
+  orderDb: OrderDb[Future],
+  assetPairBuilder: AssetPairBuilder,
+  matcherPublicKey: PublicKey,
+  matcher: ActorRef,
+  storeCommand: StoreValidatedCommand,
+  orderBook: AssetPair => Option[Either[Unit, ActorRef]],
+  orderBookHttpInfo: OrderBookHttpInfo,
+  override val matcherStatus: () => MatcherStatus,
+  override val apiKeyHash: Option[Array[Byte]]
+)(implicit mat: Materializer)
+    extends ApiRoute
+    with ProtectDirective
+    with HasStatusBarrier
+    with AuthRoute
+    with ScorexLogging {
+
+  implicit private val executionContext: ExecutionContext = mat.executionContext
+  implicit private val timeout: Timeout = settings.timeout
+
+  // NOTE: routes here must not change its places (especially getOrderBookRestrictions ~ getOrderStatusByPKAndIdWithSig ~ getOrderBook)
+  override lazy val route: Route =
+    pathPrefix("matcher") {
+      pathPrefix("orderbook") {
+        matcherStatusBarrier {
+          getOrderBookRestrictions ~ getOrderStatusByPKAndIdWithSig ~ getOrderBook ~ getOrderBooks ~ getOrderBookStatus ~ deleteOrderBookWithKey ~ getOrderStatusByAssetPairAndId
+        }
+      } ~ pathPrefix("orders") {
+        matcherStatusBarrier(getOrderStatusByAddressAndIdWithKey)
+      }
+    }
+
+  private def getOrderBookRestrictions(pair: AssetPair): FutureResult[HttpOrderBookInfo] = settings.getActualTickSize(pair).map { tickSize =>
```

Comment:
L85:99 to the bottom of the class after routes

Reply:
fixed